opentelemetry-declarative-config/src/metrics/reader_config.rs

```diff
@@ -0,0 +1,27 @@
+//! # Metrics Reader Configuration module
+//!
+//! This module defines the configuration structures for metrics readers
+//! used in OpenTelemetry SDKs. Readers are responsible for collecting
+//! metrics data and exporting it to various backends or systems.
+
+use serde::Deserialize;
+use serde_yaml::Value;
+
+/// Configuration for Metrics Readers
+#[derive(Deserialize)]
+pub struct ReaderConfig {
+    pub pull: Option<PullReaderConfig>,
+    pub periodic: Option<PeriodicReaderConfig>,
```

Comment:
We want to use an `enum` here. A `MetricReader` would either be periodic (push-based) or pull-based. It can't be both.